File: atopmall_web/goods_web/router/goods.go
package router

import (
	"atopmall_web/goods_web/api/goods"
	"atopmall_web/goods_web/middlewares"

	"github.com/gin-gonic/gin"
)

// 商品相关路由
func GoodsRouterInit(Router *gin.RouterGroup) {
	GoodsRouter := Router.Group("goods")
	{
		GoodsRouter.GET("", goods.GetGoodsList)
		GoodsRouter.POST("", middlewares.JWTAuth(), middlewares.IsAdmin(), goods.NewGoods)
		GoodsRouter.GET("/:id", goods.GetGoodsDetail)
		GoodsRouter.DELETE("/:id", middlewares.JWTAuth(), middlewares.IsAdmin(), goods.DeleteGoods)
		GoodsRouter.GET("/:id/stocks", goods.Stocks) //获取商品的库存

		//更新商品中的是否热门等信息
		GoodsRouter.PATCH("/:id", middlewares.JWTAuth(), middlewares.IsAdmin(), goods.UpdateGoodsStatus)
		GoodsRouter.PUT("/:id", middlewares.JWTAuth(), middlewares.IsAdmin(), goods.UpdateGoods)
	}
}

File: atopmall_frontend/admin/src/api/goods.ts
import request from '@/utils/request'

export interface GoodsItem {
  id: number
  name: string
  goods_sn: string
  market_price: number
  shop_price: number
  front_image: string
  goods_brief: string
  images: string[]
  desc_images: string[]
  is_hot: boolean
  is_new: boolean
  on_sale: boolean
  ship_free: boolean
  stocks: number
  brand: {
    id: number
    name: string
    logo: string
  }
  ctegory: {
    id: number
    name: string
  }
}

export interface GoodsListParams {
  p?: number
  pnum?: number
  q?: string
  c?: number
  b?: number
  pmin?: number
  pmax?: number
  ishot?: number
  isnew?: number
  istab?: number
  onsale?: number
}

export interface GoodsListResponse {
  data: GoodsItem[]
  total: number
}

export const getGoodsList = (params?: GoodsListParams) => {
  return request.get<GoodsListResponse>('/g/v1/goods', params)
}

export const getGoodsDetail = (id: number) => {
  return request.get(`/g/v1/goods/${id}`)
}

export const createGoods = (data: any) => {
  return request.post('/g/v1/goods', data)
}

export const updateGoods = (id: number, data: any) => {
  return request.put(`/g/v1/goods/${id}`, data)
}

export const updateGoodsStatus = (id: number, data: { sale: boolean; new: boolean; hot: boolean }) => {
  return request.patch(`/g/v1/goods/${id}`, data)
}

export const deleteGoods = (id: number) => {
  return request.delete(`/g/v1/goods/${id}`)
}

export const getGoodsStock = (id: number) => {
  return request.get<{ goodsId: number; num: number }>(`/g/v1/goods/${id}/stocks`)
}